site/book/05-developing-functions/03-developing-in-Go.md

```diff
@@ -1 +1,118 @@
-TODO
+You can use the `kyaml` library to development functions in Go. It provides the
+following features:
+
+- **General-purpose language:** Compared to Domain Specific Languages (DSL), Go
+  is a general-purpose programming language that provides:
+  - Proper abstractions and language features
+  - A extensive ecosystem of tooling (e.g. IDE support)
+  - A comprehensive catalog of well-supported libraries
+  - Robust community support and detailed documentation
+- **YAML-centric**: As opposed to other frameworks discussed in this chapter,
+  the `kyaml` library exposes the YAML Abstract Syntax Tree (AST) to the user.
+  This enables you to control every aspect of the YAML file including
+  manipulating comments, but comes at the cost of complexity compared to
+  representing resources as idiomatic data structures.
+
+## Quickstart
+
+### Create the go module
+
+```shell
+$ go mod init github.com/user/repo
+$ go get sigs.k8s.io/kustomize/kyaml@v0.10.6
+```
+
+### Create the `main.go`
+
+This is a simple function that adds the annotation `myannotation` with the
+provided value:
+
+```go
+// main.go
+package main
+
+import (
+	"os"
+
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+var value string
+
+func main() {
+	resourceList := &framework.ResourceList{}
+	cmd := framework.Command(resourceList, func() error {
```

Comment:
Functions should be using v0.10.19 kyaml.

https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/318

Reply:
Let's do that as part of https://github.com/GoogleContainerTools/kpt/issues/2054.

Also, as you're working with the new framework changes, think about whether we need to expand this section. It may not be sufficient in this spartan form.